src/graia/ariadne/message/parser/alconna.py

```diff
@@ -0,0 +1,423 @@
+from typing import Dict, List, Optional, Union, Type, Any, TypeVar
+from types import TracebackType
+from pydantic import BaseModel
+import re
+
+from graia.broadcast.interfaces.dispatcher import BaseDispatcher, DispatcherInterface
+
+from ..chain import MessageChain  # 请确保 MessageChain 格式的正确
+from ...event.message import MessageEvent
+
+
+class ParamsUnmatched(Exception):
+    """一个 text 没有被任何参数匹配成功"""
+
+
+def split_once(text: str, separate: str):  # 相当于另类的pop, 不会改变本来的字符串
+    out_text = ""
+    quotation_stack = []
+    is_split = True
+    for char in text:
+        if char in "'\"":  # 遇到引号括起来的部分跳过分隔
+            if not quotation_stack:
+                is_split = False
+                quotation_stack.append(char)
+            else:
+                is_split = True
+                quotation_stack.pop(-1)
+        if separate == char and is_split:
+            break
+        out_text += char
+    return out_text, text.replace(out_text, "", 1).replace(separate, "", 1)
+
+
+AnyIP = r"(\d+)\.(\d+)\.(\d+)\.(\d+)"
+AnyDigit = r"(\d+)"
+AnyStr = r"(.+)"
+AnyUrl = r"(http[s]?://.+)"
+
+NonTextElement = TypeVar("NonTextElement")
```

Comment:
this should be `Element` in graia.ariadne.message.element
like 
```python
from ..element import Element
NonTextElement = TypeVar("NonTextElement", bound=Element)
```
or replace `NonTextElement`  to `Element`